### PR TITLE
Support pre-commit from inside submodules

### LINKED ISSUE #213: Does not work within submodules
I'm getting: 
```
An unexpected error has occurred: NotADirectoryError: [Errno 20] Not a directory: '/home/quentin/chef-repo/cookbooks/ssmtp-cookbook/.git/hooks/pre-commit'
```
chef-repo is my primary repository and ssmtp-cookbook a git submodule of that. 

**ssmtp-cookbook/.git file contents:**
```
gitdir: ../../.git/modules/cookbooks/ssmtp-cookbook
```

### PR DESCRIPTION
**Pull request by asottile**
Resolves #213

### CODE REVIEW

File: pre_commit/main.py [hunk 1]
```diff
@@ -25,6 +25,13 @@
 # https://github.com/pre-commit/pre-commit/issues/300
 # In git 2.6.3 (maybe others), git exports this while running pre-commit hooks
 os.environ.pop('GIT_WORK_TREE', None)
+# In git 1.9.1 (maybe others), git exports these while running pre-commit hooks
+# in submodules.  In the general case this causes problems.
+# These are covered by test_install_in_submodule_and_run
+# Causes git clone to clone wrong thing
+os.environ.pop('GIT_DIR', None)
```
Comment (chriskuehl) [inline pre_commit/main.py]: lol

File: pre_commit/main.py [hunk 2]
```diff
@@ -25,6 +25,13 @@
...
+# Causes 'error invalid object ...' during commit
+os.environ.pop('GIT_INDEX_FILE', None)
```
Comment (chriskuehl) [inline pre_commit/main.py]: more lol

Comment (asottile) [inline pre_commit/main.py]: Fuckin right?

Comment (asottile) [inline pre_commit/main.py]: Why are submodules so different and complicated and broken

**Pull request merged**